source/common/tcp_proxy/upstream.cc

```diff
@@ -11,6 +12,24 @@
 
 namespace Envoy {
 namespace TcpProxy {
+
+const std::vector<std::reference_wrapper<const Router::RateLimitPolicyEntry>>
```

Comment:
sorry to keep coming back to this but if all we're doing is combining the http1 and http2 upstreams into one class I don't think we should need all of this
/wait

Reply:
This new class owns a `Router::UpstreamRequest` instance. Unlike existing `Http1Upstream` and `Http2Upstream`, new class handles traffic generically by invoking appropriate functions on `Router::UpstreamRequest`. To test these methods, tests need to initialize `Router::UpstreamRequest` and therefore now test are using `NullRouteImpl` lib.